common/bl_common.c

```diff
@@ -101,6 +101,7 @@ int is_mem_free(uintptr_t free_base, size_t free_size,
 	return (addr >= free_base) && (requested_end <= free_end);
 }
 
+#if defined(IMAGE_BL1)
```

Comment:
This is also used by BL2, as far as I know. I think that what we should do is remove this file from the list of common files in the Makefile (```BL_COMMON_SOURCES```) and move it to the makefiles of BL1 and BL2 (```BL1_SOURCES``` and ```BL2_SOURCES```). To make sure that you haven't broken anything just run:
```
CROSS_COMPILE=aarch64-linux-gnu- make -j all PLAT=fvp DEBUG=1
```
If this builds, it should be fine.

Reply:
I dont think that will work as other functions from this file are required for BL31